packages/@aws-cdk/aws-apigateway/README.md

```diff
@@ -582,6 +582,36 @@ OPTIONS added to them.
 See [#906](https://github.com/aws/aws-cdk/issues/906) for a list of CORS
 features which are not yet supported.
 
+### Endpoint Configuration
+To define an endpoint type for the API gateway, use `endpointConfiguration` property:
+
+```ts
+const api = new apigw.RestApi(stack, 'api', {
+  endpointConfiguration: {
+    types: [ apigw.EndpointType.EDGE, apigw.EndpointType.PRIVATE ]
+  }
+});
+```
+
+You can also create an association between your Rest Api and a Vpc Endpoint. By doing so,
+Api Gateway will generate a new Route53 Alias DNS record which you can use to invoke your private
+APIs. Here is an example:
+
+```ts
+const someEndpoint: IVpcEndpoint = /* Get or Create endpoint here */
+const api = new apigw.RestApi(stack, 'api', {
+  endpointConfiguration: {
+    types: [ apigw.EndpointType.PRIVATE ],
+    vpcEndpoints: [someEndpoint]
+  }
+});
+```
+
+By performing this association, we can invoke the api gateway using the following format:
+```
+https://{rest-api-id}-{vpce-id}.execute-api.{region}.amazonaws.com/{stage}
+```
```

Comment:
How hard is it to add an integration test that sets up a simple VPC endpoint and associates it to the Rest API, such that it is invokable by running `curl` against `https://{rest-api-id}-{vpce-id}.execute-api.{region}.amazonaws.com/{stage}`?

This would be useful as [described here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md#integration-tests) as well as to link off from the documentation as a working example.

Reply:
Thanks for the feedback. No problem at all. I will go ahead and add. It sounds like `Step verification steps:` is the part where I can add the `curl` commands?

Reply:
Sorry for the late reply. Yes, that's correct, Manual verification is fine.